test/test_jit.py

```diff
@@ -17636,6 +17636,15 @@ def _forward(self, x):
         input = torch.ones(2, 2)
         self.assertEqual(input, sm(input))
 
+    # Tests the case where a torch.Tensor subclass is used as input.
+    def test_script_module_tensor_subclass_argument(self):
```

Comment:
There is one more case that we should test, which is using a `Parameter` annotation in the `__init__` function of a `Module` to initialize a `Parameter` attribute.

Reply:
Done, added a case testing that.

Reply:
I did some more investigation looks like type annotations are not used in this case to infer the type. As we discussed, parameters are special and automatically get added to the `_parameters` attribute of the module, so there's no need to try and guess the type. 

Can you revert this extra test? Sorry about this!

Reply:
Np. reverted.